### PR TITLE
Explosion Audio

### PR DESCRIPTION
**Pull request by NanakoAC**
Adds directional audio for explosions and fixes a variety of sound issues with them.
Adds screenshaking for explosions
Tweaks volumes and adds missing sounds for several window/airlock interactions.

Also fixes some runtime errors here and there which were discovered during testing, mostly just adding safeties

### CODE REVIEW

**Comment by NanakoAC** [comment]
the changes to mulebots were mentioned implicitly - it was part of fixing miscellaneous runtime errors. their ex_act would throw a runtime error the way it was before, so i altered it to do something more or less the same as the original intent, that actually works

File: code/game/objects/explosion.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
 		if(z_transfer && (devastation_range > 0 || heavy_impact_range > 0))
 			//transfer the explosion in both directions
 			explosion_z_transfer(epicenter, devastation_range, heavy_impact_range, light_impact_range, flash_range)
-///// Z-Level Stuff
+///// Z-Level Stuffis
```
Comment (NanakoAC) [inline code/game/objects/explosion.dm]: clearly a typo, cursor was probably in the wrong place when i typed something. i have no memory of this little change

File: code/game/objects/explosion.dm [hunk 2]
```diff
@@ -31,33 +31,33 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
...
-						far_volume += (dist <= far_dist * 0.5 ? 50 : 0) // add 50 volume if the mob is pretty close to the explosion
-						M.playsound_local(epicenter, 'sound/effects/explosionfar.ogg', far_volume, 1, frequency, falloff = 5)
-
-		var/close = range(world.view+round(devastation_range,1), epicenter)
-		// to all distanced mobs play a different sound
-		for(var/mob/M in world) if(M.z == epicenter.z) if(!(M in close))
-			// check if the mob can hear
-			if(M.ear_deaf <= 0 || !M.ear_deaf) if(!istype(M.loc,/turf/space))
-				M << 'sound/effects/explosionfar.ogg'
+					if (M.ear_deaf <= 0 || !M.ear_deaf)//Dont play sounds to deaf people
```
Comment (skull132) [inline code/game/objects/explosion.dm]: Missing a check for the user being in space.

Sound requires a medium to travel by, and being in space should mean you don't hear the boom. This was covered by the `if(!istype(M.loc,/turf/space)` condition in the outdated version.

**Unless** `playsound_local()` takes care of it.

File: code/game/objects/explosion.dm [hunk 3]
```diff
@@ -114,7 +114,8 @@ proc/explosion(turf/epicenter, devastation_range, heavy_impact_range, light_impa
 
 	return 1
 
-
+/mob/verb/camerashake(strength as num)
```
Comment (skull132) [inline code/game/objects/explosion.dm]: This being a verb sounds like a mistake. A verb is something any player can initiate: players don't have a use for this.

Also, can't you find a better place to shove this into? `code/modules/mob/mob_helpers.dm` smells like a great place.

File: code/game/objects/items/devices/transfer_valve.dm [hunk 4]
```diff
@@ -165,18 +165,18 @@
 /obj/item/device/transfer_valve/proc/split_gases()
 	if(!valve_open)
 		return
-	
+
 	valve_open = 0
-	
-	if(deleted(tank_one) || deleted(tank_two))
+
+	if(deleted(tank_one) || deleted(tank_two) || !tank_one.air_contents || !tank_two.air_contents)
```
Comment (skull132) [inline code/game/objects/items/devices/transfer_valve.dm]: I assume these were causing runtimes?

File: code/game/objects/explosion.dm (hunk 3 above)
Comment (NanakoAC) [inline code/game/objects/explosion.dm]: oops this was actually just for debugging, ill remove it

File: code/game/machinery/doors/door.dm [hunk 5]
```diff
@@ -330,9 +338,9 @@
 		if(W.damtype == BRUTE || W.damtype == BURN)
 			user.do_attack_animation(src)
 			if(W.force < min_force)
-				user.visible_message("\red <B>\The [user] hits \the [src] with \the [W] with no visible effect.</B>" )
+				user.visible_message(span("danger","\The [user] hits \the [src] with \the [W] with no visible effect.</B>" ))
```
Comment (skull132) [inline code/game/machinery/doors/door.dm]: Unpaired `</B>` token. One on line 343 too.

**Comment by NanakoAC** [comment]
yes no sounds in space.  HTML error fixed too this should be done

**Pull request merged**